site/scripts/db-indexes.ts

```diff
@@ -0,0 +1,55 @@
+/* eslint-disable import/first */
+import dotenv from "dotenv";
+
+dotenv.config({ path: ".env.local" });
```

Comment:
Let’s use [`dotenv-flow`](https://www.npmjs.com/package/dotenv-flow) instead. It matches [how Next.js reads env vars](https://nextjs.org/docs/basic-features/environment-variables).

Reply:
We can also avoid `eslint-disable` I guess. To do this, we can start the script with `import "../shared/environment"`, which will in turn call:

```ts
import { config } from "dotenv-flow";

export const monorepoRootDir = path.resolve(__dirname, "../../..");
config({ silent: true, path: monorepoRootDir });
```

Another way of solving this can be adding `exe` to `package.json` → `scripts`:

```
  "exe": "ts-node --require dotenv-flow/config --transpile-only"
```

Then we can `yarn exe scripts/a/b/c`.

Reply:
Adding `yarn exe` here: https://github.com/blockprotocol/blockprotocol/pull/169

Reply:
How should we handle the existing `scripts` in the site project, then? 
Should we remove them and put them at the root package.json using `exe`, or keep them as is and make naming more consistent?

Reply:
That’s a good question! Both options have pros and cons.

**If all scripts are in monorepo root**

- 👍 There is only one `yarn exe` and one `/scripts/` folder
- 👍 Commands are shorter (`yarn exe scripts/a/b/c` instead of `yarn workspace package-name exe scripts/a/b/c`)
- 👍 Shell autocompletion is more likely to work for script paths because they are relative to CWD
- 👍 We don’t need to exclude `packages/a/scripts` from publishing (I assume scripts are only for repo maintenance, they are not the same as publishable CLI commands in NPM packages).
- 🤔 We have to keep more `devDependencies` in root `package.json`; those dependencies will be quite mixed
- 🤔 If `devDependencies` are quite big (e.g. Puppeteer), it’s harder to do selective installs (e.g. to build a docker image out of a couple of packages). Not sure we need to think about this in this repo, the question is more related to [HASH](https://github.com/hashintel/hash).

**If all scripts are in individual packages**

Same as above, but with 👍 and 🤔 swapped. Plain count of pros and cos does not mean that some option is better.

I once worked on a monorepo where all scripts were put into `packages/our-scripts`, so they had their own `package.json`. It had cons of both options, so I can’t say it was any better 😅

To be fair, it should be quite simple to jump between global and package-scoped scripts. We can try any option in a consistent way, see how it works for us and then swap the approach if we are not happy 🙂